packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -1114,7 +1124,11 @@ class ListView extends BoxScrollView {
     ScrollViewKeyboardDismissBehavior keyboardDismissBehavior = ScrollViewKeyboardDismissBehavior.manual,
     String? restorationId,
     Clip clipBehavior = Clip.hardEdge,
-  }) : childrenDelegate = SliverChildListDelegate(
+  }) : assert(
+         !(itemExtent != null && prototypeItem != null),
```

Comment:
this might be easier to read as `itemExtent == null || prototypeItem == null`

Reply:
(this applies in other cases below also)

Reply:
@Hixie I see, I was going to do this way, but I then decided to follow the pattern that I found in some places for excluding parameters.

Refert to `InputDecoration`, `Material`, `PopupMenuButton`, `ShapeDecoration` and `AsyncSnapshot`.

Reply:
Thinking better, I don't think they are equivalent as, if one of them is `null` the error will be thrown, even if the idea is to allow both being `null`.